components/browser/session/src/main/java/mozilla/components/browser/session/storage/SessionStorage.kt

```diff
@@ -4,28 +4,39 @@
 
 package mozilla.components.browser.session.storage
 
+import mozilla.components.browser.session.Session
 import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+
+data class SessionWithState(
+    val session: Session,
+    val selected: Boolean = false,
+    val engineSession: EngineSession? = null
+)
+typealias SessionsSnapshot = List<SessionWithState>
```

Comment:
I wonder why you added the `selected` Boolean to all `SessionWithState` objects instead of turning `SessionsSnapshot` into a class that can hold global values too. Maybe there are going to be more. :)